src/main.rs

```diff
@@ -123,6 +125,17 @@ fn check_cache(args: &Args, cache: &Cache) {
     if !args.flag_update {
         match cache.last_update() {
             Some(ago) if ago > MAX_CACHE_AGE => {
+                if args.flag_auto_update {
+                    cache.update().unwrap_or_else(|e| {
+                        match e {
+                            CacheError(msg) | ConfigError(msg) | UpdateError(msg) => {
+                                eprintln!("Could not update cache: {}", msg)
+                            }
+                        };
+                    });
+                    return;
+                }
```

Comment:
I think it would be good to print an information if the cache was successfully updated:

```rust
if !args.flag_quiet {
    println!("Successfully updated cache.");
}
```

Reply:
Yes, but since I don't abort with the '-a' flag when updating fails I'll need to do it with a `.map` call.

Reply:
Yes, or match :)

Reply:
I did it with map, didn't see your comment.

Reply:
To be honest `match` would be much more readable in this case. Don't you think?

Reply:
Well it looks almost exactly the same, except that it uses on more line:

```diff
diff --git a/src/main.rs b/src/main.rs
index 616829a..d21da36 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -126,19 +126,20 @@ fn check_cache(args: &Args, cache: &Cache) {
         match cache.last_update() {
             Some(ago) if ago > MAX_CACHE_AGE => {
                 if args.flag_auto_update {
-                    cache.update()
-                        .map(|()| {
+                    match cache.update() {
+                        Ok(()) => {
                             if !args.flag_quiet {
                                 println!("Successfully updated cache.");
                             }
-                        })
-                        .unwrap_or_else(|e| {
-                            match e {
+                        }
+                        Err(err) => {
+                            match err {
                                 CacheError(msg) | ConfigError(msg) | UpdateError(msg) => {
                                     eprintln!("Could not update cache: {}", msg)
                                 }
                             };
-                        });
+                        }
+                    }
                     return;
                 }
```

Reply:
What about this?

```rust
match cache.update() {
    Ok(_) if !args.flag_quiet => println!("Successfully updated cache."),
    Ok(_) => {},
    Err(e) => match e {
        CacheError(msg) | ConfigError(msg) | UpdateError(msg) => {
            eprintln!("Could not update cache: {}", msg)
        }
    },
}
```

(The error match pattern could also be collapsed, but then it's a bit repetitive. There's an RFC to allow things like `Err(CacheError(msg) | ConfigError(msg) | UpdateError(msg))` in the future, but it hasn't landed yet AFAIK.)

Reply:
Well, you argued with length, and this version is shorter. And I *love* my bikeshed in darker hues of blue! Light blue is just abhorrent...

Reply:
And if you prefer it, you can also use this:

```rust
match cache.update() {
    Ok(_) => if !args.flag_quiet => {
        println!("Successfully updated cache.");
    },
    Err(e) => match e {
        CacheError(msg) | ConfigError(msg) | UpdateError(msg) => {
            eprintln!("Could not update cache: {}", msg)
        }
    },
}
```

I think that's actually the nicest version :slightly_smiling_face:

Reply:
@dbrgn Ping. I applied your favorite shade of medium blue :wink:

Reply:
Sorry, I've been sick for the last three days. Will take a look in a few days.